### PR TITLE
fix(win32): keep the tab tooltip up while its title animates

### PR DESCRIPTION
**Pull request by nanasess**
## Summary

The tab title tooltip from #230 flickers over a tab whose title animates, and disappears for good if the pointer holds still.

`syncTabButtons` hid the tooltip whenever a label rebuild touched the tab it was up for ("hides on ... label rebuild" in #230). That suits a rename, but Claude Code spins a glyph through the OSC title while it works, so the label is rebuilt on every frame: the popup went down each time, and the next `WM_MOUSEMOVE` re-armed `TME_HOVER` and brought it back after the hover delay. With the pointer moving even slightly that is a flicker for as long as it rests on the tab; with the pointer perfectly still nothing re-arms the hover and the tooltip never comes back.

This PR:

- makes the rebuild mark the tooltip stale and re-present it in place once `layoutChromeForRect` has settled the button rects, since the popup hangs off the button's final rect: the text is replaced, the popup re-measured and moved, and a title that no longer needs compacting still takes it down.
- splits `showTabTooltip` so the hover path and the new `refreshTabTooltip` share `presentTabTooltip`.
- checks that the pointer is still over the button before re-presenting. A relayout can move the tab out from under a pointer that has not moved, and the button's `WM_MOUSELEAVE` for that arrives on the tracker's own clock, so without the check the popup could be re-shown under a tab nobody points at.
- raises UIA `NameChanged` only when the title itself changed, not when only the width budget did (`tabTooltipUiaName` reports the full title, which is unchanged in that case).

## Validation

- [x] `zig build -Demit-exe=true -Doptimize=ReleaseSafe --summary all` — 76/76 steps
- [x] `zig build test -Dtest-filter=ConPTY -Dtest-filter=tooltip -Dtest-filter=label -Dtest-filter=tab --summary all` — 511/515 passed, 4 skipped
- [x] `zig build test -Demit-test-exe=true --summary all` — 4409/4487 passed, 78 skipped, 0 failed
- [x] `zig fmt --check src/apprt/win32.zig` — clean (`scripts/check-source-format.ps1` only covers `.ps1` / `.psm1` / `.json`, none touched)
- [x] Manual, Windows 11, ReleaseSafe with the bundled ConPTY, WSL shell running Claude Code in a tab whose label is compacted. Before: hovering the tab while Claude Code animates the title shows the tooltip blinking in and out on every title change (screen recording). After: the tooltip stays up and its text follows the title.

## Risks / Follow-ups

- The popup is re-measured on every title change, so a spinner whose glyphs differ in advance width moves the popup's right edge by a few px per frame. Not noticeable in the manual check; if it bothers anyone, a width that only grows while the tab stays the same would be the follow-up.
- The rename and relayout paths (tab added or removed while the tooltip is up, title shrinking back under the budget) go through the same `presentTabTooltip` and were reasoned through rather than exercised one by one in the manual check.
- No new unit test: the changed logic is bound to the HWNDs (`GetWindowRect`, `GetCursorPos`, the popup itself), and `win32_tab_tooltip.zig` keeps covering the placement math untouched.

Per `AI_POLICY.md`: the investigation, the change and this description were done with Claude Code. I reviewed the diff, ran the validation above and did the manual check myself.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Keep Windows tab title tooltips stable and correctly positioned during title animation and tab layout changes.

Bug Fixes:
- Keep tab title tooltips visible and synchronized while animated titles update, while hiding them when the pointer no longer targets the tab or the tab becomes hidden.

Enhancements:
- Avoid redundant tooltip repainting and UIA NameChanged notifications when only layout or display text changes without changing the full title.
- Refresh tooltip placement after tab layout changes so it follows moved or resized tab buttons.

Tests:
- Add coverage for tooltip presentation with hidden buttons and pointer boundary conditions.

Chores:
- Document edge cases involving hidden tab buttons and tooltip refresh behavior.

## Summary by CodeRabbit

* **Bug Fixes**
  * Tab tooltips now refresh in place when tab titles or available label space changes.
  * Stale tooltips are hidden when the pointer leaves the tab or the active tab can no longer be identified.
  * Tooltips are shown only for visible tabs under the pointer.
  * Unchanged tooltip text no longer triggers unnecessary visual or accessibility updates.
  * Tooltip display remains synchronized after tab layout changes.
  * Accessibility name updates now trigger reliably when a tab title changes.

### CODE REVIEW

**Comment by amanthanvi** [comment]
Thanks @nanasess, this is a good fix and the measurement agrees with your description. With a pwsh child rewriting the OSC title every 100 ms and the real pointer inside the first tab button, main's popup was up for 90 ms of a 2.7 s dwell and toggled twice; your head kept it up for the whole dwell with the text following the title.

I pushed one follow-up commit to your branch (4bb385fe4) rather than asking for another round, since it needed a scenario that is fiddly to set up:

- **A tab that scrolls out of the strip kept its tooltip.** `layoutChromeForRect` hides a button that left `visibleTabRange` with `ShowWindow` alone, so `GetWindowRect` still returned its old slot and the pointer check passed against whatever tab now sits there. Measured with 14 tabs in a 60-column window: tooltip up over tab 11, next-tab past the end wraps the range to tabs 1 through 4, and the popup for tab 11 stayed up over tab 1. The check now goes through a pure `mayPresent` in `win32_tab_tooltip.zig` (with tests) that also takes the strip's own visibility record plus `shouldShowTabBar()`, and `syncTabButtons` refreshes whenever a tooltip is up rather than only when its own tab's label was rebuilt, because the tab that scrolls out because another tab was activated is never rebuilt.
- **A refresh that changes nothing is now a no-op.** The popup remembers its text, so an animated title only raises UIA `NameChanged` and repaints when the string really moved, and a width-budget change that left the full title alone announces nothing.

Tests: `zig build test -Dtest-filter=ConPTY -Dtest-filter=tooltip -Dtest-filter=tab -Dtest-filter=label --summary all` is 513/517 passed, 4 skipped. Will squash-merge once CI and the review bots are through, with you as author.

---
*This comment was written by an AI agent (Claude Code) operating the maintainer's account, with the maintainer's authorization. The maintainer reviewed the follow-up diff and the measurements.*

File: src/apprt/win32.zig [hunk 1]
```diff
@@ -13996,10 +14052,17 @@ const Host = struct {
...
+        // for a title the popup already displays.
+        const text_changed = !ownedStringEquals(self.tab_tooltip_text, text);
+        appendOwnedString(alloc, &self.tab_tooltip_text, text) catch return;
         _ = sys.SetWindowTextW(tooltip, text_w.ptr);
-        const tooltip_tab_changed = self.tab_tooltip_tab_id != tab.id;
         self.tab_tooltip_tab_id = tab.id;
-        if (tooltip_tab_changed) {
+        if (name_changed and text_changed) {
             if (self.tab_tooltip_uia_provider) |provider| provider.raiseNameChanged();
         }
```
[the remark this answers is not in the document]
Comment (amanthanvi) [reply src/apprt/win32.zig:14069]: Good catch, and the fix is in 9e4a93f9d. `buildTabTooltipText` bounds the drawn text by `tab_tooltip_max_width` while `tabTooltipUiaName` reports the full title, so a title that changed past that bound renders identically and still has to announce. `NameChanged` is now gated on `name_changed` alone, as your suggestion shows; the remembered text only decides whether to repaint.

---
*Reply written by an AI agent (Claude Code) operating the maintainer's account, with the maintainer's authorization.*

File: src/apprt/win32.zig [hunk 2]
```diff
@@ -13996,10 +14052,17 @@ const Host = struct {
...
+        // re-announce the name to a reader or re-place the popup. The text
+        // is compared as the popup holds it, so this also covers a refresh
+        // for a title the popup already displays.
+        const text_changed = !ownedStringEquals(self.tab_tooltip_text, text);
+        appendOwnedString(alloc, &self.tab_tooltip_text, text) catch return;
         _ = sys.SetWindowTextW(tooltip, text_w.ptr);
-        const tooltip_tab_changed = self.tab_tooltip_tab_id != tab.id;
         self.tab_tooltip_tab_id = tab.id;
-        if (tooltip_tab_changed) {
+        if (name_changed and text_changed) {
```
[the remark this answers is not in the document]
Comment (amanthanvi) [reply src/apprt/win32.zig]: Agreed, the comment overstated it. Reworded in 8be757c1c: an unchanged-text refresh still re-places the popup against the button's final rect and only skips the name announcement and the repaint.

---
*Reply written by an AI agent (Claude Code) operating the maintainer's account, with the maintainer's authorization.*

**Comment by amanthanvi** [comment]
Merged as 627d67d99, with you as author. Thanks again for the fix and the careful write-up; it will be in the next release.

---
*This comment was written by an AI agent (Claude Code) operating the maintainer's account, with the maintainer's authorization.*
